src/java/org/apache/cassandra/io/util/DataOutputBuffer.java

```diff
@@ -221,6 +240,17 @@ public ByteBuffer buffer(boolean duplicate)
         return result;
     }
 
+    /**
+     * Gets the underlying ByteBuffer and calls {@link ByteBuffer#flip()}.  This method is "unsafe" in the sense that
+     * it returns the underlying buffer, which may be modified by other methods after calling this method.  If the calling
+     * logic knows that no new calls to this object will happen after calling this method, then this method can avoid the
+     * copying done in {@link #getData()} and {@link #asNewBuffer()}.
```

Comment:
nit: I guess the `asNewBuffer()` part is no longer relevant, as we're sticking with that in `truncationAsMapEntry()`?

Reply:
I think the original intent was also to avoid `buffer(true)`, which we no longer used outside `HintsWriter`.

Reply:
`asNewBuffer` is still used in Describe and SystemKeyspace

`buffer` I feel is newer than when I first wrote it... there are duplicate methods in this class huh !

Reply:
added buffer to the doc